constraint/pkg/regorewriter/regorewriter.go

```diff
@@ -124,7 +124,17 @@ func (r *RegoRewriter) addFileFromFs(path string, slice *[]*Module) error {
 		return fmt.Errorf("%w: %v", ErrReadingFile, err)
 	}
 
-	m, err := ast.ParseModule(path, string(bytes))
+	attemptVersions := []ast.RegoVersion{ast.RegoV0, ast.RegoV1}
```

Comment:
why is this using v0 first then v1 while https://github.com/open-policy-agent/frameworks/pull/517/files#diff-f21596183dbd319da5a0fe99596c9a346cc60a7222b062b6beb5342a74242724R242 in compilers.go parses to v1 first then v0? Was this intentional? If so, maybe add a comment to explain the rationale?

Since no version in CT means v0, does it make sense to default to v1 always and fallback to v0?

Reply:
Good spot, I'd missed updating this to be the same order as parsing in 947c51b957eddd426fd0cc27f4cbcf67751ef26b.

> Since no version in CT means v0

I was under the impression we would add the version field in another PR. But I can add it in this PR instead if that's preferred. I think if we have versioned CRs, then we might as well drop this multi-attempt parsing functionality. This was intended just to be an interim option to allow folk to use v1 Rego before we had controllable versioning. Lmk what you prefer and I can do that.

Reply:
Based on past discussions, we want users to opt-in to rego v1 by explicitly adding:
```yaml
  targets:
    - target: admission.k8s.gatekeeper.sh
      code:
      - engine: Rego
        source:
          version: v1
          rego: |
```
When they don't provide version, then we assume v0 to ensure backward compatibility.

Given this, should we default to parse v0 first, and if failed, then parse with v1?

Reply:
Ahh, ok I can get that done in this PR instead. (https://github.com/open-policy-agent/frameworks/pull/517#issuecomment-2680124873)

In which case, I will remove the attemptVersions parsing and only parse v0 unless the source.version field is set to v1. 

I can get that done this week.

Reply:
Hey, I have just pushed 37bcfcbe6c038382d4ab1f3971cd99ee121579bd which does this.